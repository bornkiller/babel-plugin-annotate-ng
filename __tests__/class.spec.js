/**
 * @description - babel-plugin-annotate-ng unit test
 * @author - bornkiller <hjj491229492@hotmail.com>
 */

const path = require('path');
const babel = require('babel-core');
const annotate = require('../src');
// fixture
const babelOptions = {
  presets: [],
  plugins: [annotate],
  babelrc: false
};

describe('babel-plugin-annotate-ng Class Declare', function () {
  it('ExportNamedDeclare', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Class/redux.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });

  it('ExportDefaultDeclaration', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Class/monitor.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });
});

describe('babel-plugin-annotate-ng Inline Class Declare', function () {
  it('ExportNamedDeclare', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Class/redux-inline.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });

  it('ExportDefaultDeclare', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Class/monitor-inline.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });
});

describe('babel-plugin-annotate-ng Cross Class Declare', function () {
  it('ExportNamedDeclare', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Class/redux-cross.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });

  it('ExportDefaultDeclare', function () {
    const fixture = path.resolve(__dirname, '../__fixture__/Class/monitor-cross.controller.js');
    const { code } = babel.transformFileSync(fixture, babelOptions);

    expect(code).toMatchSnapshot();
  });
});
