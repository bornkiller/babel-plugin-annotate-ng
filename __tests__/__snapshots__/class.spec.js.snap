// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-annotate-ng Class ExportDefaultDeclaration 1`] = `
"MonitorController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];
export default class MonitorController {
  constructor($q, $sce, $animate) {
    this.$q = $q;
    this.$sce = $sce;
    this.$animate = $animate;
  }

  ngOnInit() {}

  ngAfterViewInit() {}

  ngOnDestroy() {}
}"
`;

exports[`babel-plugin-annotate-ng Class ExportNamedDeclaration 1`] = `
"ReduxCoreController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];
export class ReduxCoreController {
  constructor($q, $sce, $animate) {
    this.$q = $q;
    this.$sce = $sce;
    this.$animate = $animate;
  }

  ngOnInit() {}

  ngAfterViewInit() {}

  ngOnDestroy() {}
}"
`;

exports[`babel-plugin-annotate-ng Cross Class Declare ExportDefaultDeclare 1`] = `
"MonitorCrossController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];
export default class MonitorCrossController {
  constructor($q, $sce, $animate) {
    this.$q = $q;
    this.$sce = $sce;
    this.$animate = $animate;
  }

  ngOnInit() {}

  ngAfterViewInit() {}

  ngOnDestroy() {}
}"
`;

exports[`babel-plugin-annotate-ng Cross Class Declare ExportNamedDeclare 1`] = `
"ReduxCrossController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];
export class ReduxCrossController {
  constructor($q, $sce, $animate) {
    this.$q = $q;
    this.$sce = $sce;
    this.$animate = $animate;
  }

  ngOnInit() {}

  ngAfterViewInit() {}

  ngOnDestroy() {}
}"
`;

exports[`babel-plugin-annotate-ng Inline Class ExportDefaultDeclaration 1`] = `
"MonitorInlineController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];
export default class MonitorInlineController {
  constructor($q, $sce, $animate) {
    this.$q = $q;
    this.$sce = $sce;
    this.$animate = $animate;
  }

  ngOnInit() {}

  ngAfterViewInit() {}

  ngOnDestroy() {}
}"
`;

exports[`babel-plugin-annotate-ng Inline Class ExportNamedDeclaration 1`] = `
"ReduxCoreController.$inject = [\\"$q\\", \\"$sce\\", \\"$animate\\"];
export class ReduxCoreController {
  constructor($q, $sce, $animate) {
    this.$q = $q;
    this.$sce = $sce;
    this.$animate = $animate;
  }

  ngOnInit() {}

  ngAfterViewInit() {}

  ngOnDestroy() {}
}"
`;
